modules/c/pages/working-with-vector-search.adoc

```diff
@@ -0,0 +1,230 @@
+
+NOTE: The number of vectors, the width or dimensions of the vectors and the training size can incur high CPU and memory costs as the size of each variable increases.
+This is because the training vectors have to be resident on the machine.
+
+=== Vector Index Configuration
+
+The table below displays the different configurations you can modify within your `VectorIndexConfiguration()` function.
+For more information on specific configurations, see xref:c:vector-search.adoc[Vector Search.]
+
+IMPORTANT: The beta release currently only supports the format of vectors as an array of numbers.
+
+.Vector Index Configuration Options
+[cols ="4*"]
+|===
+|Configuration Name |Is Required |Default Configuration |Further Information
+
+|Expression
+|Required
+|No default
+|A SQL++ expression indicating where to get the vectors. 
+A document property for embedded vectors
+`prediction()` to call a registered Predictive model.
+|Expression Language
+|Required
+|No default
+|Specifies the query language used. This can be SQL++ or N1Q1 query.
+|Number of Dimensions
+|Required
+|No default
+|2-2048
+|Number of Centroids
+|Required
+|No default
+|1-64000. The general guideline is an approximate square root of the number of documents
+|Distance Metric
+|Optional
+|Euclidean Distance
+|You can also configure Cosine Distance as your Distance Metric
+|Encoding
+|Optional
+| Scalar Quantizer(SQ) or SQ-8 bits
+a|There are three possible configurations:
+
+* None
+No compression, No data loss
+* Scalar Quantizer (SQ) or SQ-8 bits (Default)
+Reduces the number of bits per dimension 
+* Product Quantizer (PQ)
+Reduces the number of dimensions and bits per dimension
+
+|Training Size
+|Optional
+a|There are defaults for both the minimum and maximum training size for training the vectors
+
+* The minimum training size is set to 25x the number of Centroids
+
+* The maximum training size is set to 256x the number of Centroids
+|
+
+|===
+
+CAUTION: Altering the default training sizes could be detrimental to the accuracy of returned results produced by the model and total computation time.
+
+== Generating Vectors
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
+
+. You can use the `prediction()` function to generate vectors to be indexed for each document at the indexing time.
+
+Below are example configurations of the previously mentioned methods.
+
+=== Create a Vector Index with Embeddings
+
+This method shows you how to create a Vector Index with embeddings.
+
+[source, c]
+----
+
+include::c:example$code_snippets/VectorSearch.h[tags=vs-create-index]
+
+----
+
+. First, create the standard configuration, setting up an expressions, number of dimensions and number of centroids for the vector embedding.
+
+. Next, create a collection variable passing it the contents of a given collection from our `database` object, in our case, the `colors` collection.
```

Comment:
Next, get the `colors` collection from the database.

Reply:
Much more to the point, nice! Changed